duration till next ticket price

song50119 opened the pull request:
Resolves #480

Commit:
duration till next ticket price

Sirmorrison commented on ui/load/load.go:
```diff
@@ -55,7 +55,7 @@ type Icons struct {
 	WatchOnlyWalletIcon, CurrencySwapIcon, SyncingIcon, ProposalIconActive, ProposalIconInactive,
 	Restore, DocumentationIcon, DownloadIcon, TimerIcon, TicketIcon, TicketIconInactive, StakeyIcon,
-	List, ListGridIcon, DecredSymbolIcon *widget.Image
+	List, ListGridIcon, DecredSymbolIcon, TimeIcon *widget.Image
```
Kindly remove the `TimeIcon` we already have a `TimerIcon`

Sirmorrison commented on ui/load/load.go:
```diff
@@ -169,6 +169,7 @@ func NewLoad(th *decredmaterial.Theme, decredIcons map[string]image.Image) *Load
 		List:                       &widget.Image{Src: paint.NewImageOp(decredIcons["list"])},
 		ListGridIcon:               &widget.Image{Src: paint.NewImageOp(decredIcons["list_grid"])},
 		DecredSymbolIcon:           &widget.Image{Src: paint.NewImageOp(decredIcons["decred_symbol"])},
+		TimeIcon:                   &widget.Image{Src: paint.NewImageOp(decredIcons["time"])},
```
Kindly remove the TimeIcon we already have a TimerIcon

Sirmorrison commented on ui/page/tickets/utils.go:
```diff
@@ -386,3 +387,34 @@ func createClickGestures(count int) []*gesture.Click {
+	if allsecs == 0 {
+		return "imminent"
+	}
+	str := ""
+	if allsecs > 604799 {
+		weeks := allsecs / 604800
+		allsecs %= 604800
+		str += fmt.Sprintf("%dw ", weeks)
+	}
+	if allsecs > 86399 {
+		days := allsecs / 86400
+		allsecs %= 86400
+		str += fmt.Sprintf("%dd ", days)
+	}
+	if allsecs > 3599 {
+		hours := allsecs / 3600
+		allsecs %= 3600
+		str += fmt.Sprintf("%dh ", hours)
+	}
+	if allsecs > 59 {
+		mins := allsecs / 60
+		allsecs %= 60
+		str += fmt.Sprintf("%dm ", mins)
+	}
+	if allsecs > 0 {
+		str += fmt.Sprintf("%ds ", allsecs)
+	}
+	return str
+}
```
```suggestion
	var str string
	if allsecs > 604799 {
		weeks := allsecs / 604800
		allsecs %= 604800
		str = fmt.Sprintf("%dw ", weeks)
	}
	if allsecs > 86399 {
		days := allsecs / 86400
		allsecs %= 86400
		str = fmt.Sprintf("%dd ", days)
	}
	if allsecs > 3599 {
		hours := allsecs / 3600
		allsecs %= 3600
		str = fmt.Sprintf("%dh ", hours)
	}
	if allsecs > 59 {
		mins := allsecs / 60
		allsecs %= 60
		str = fmt.Sprintf("%dm ", mins)
	}
	if allsecs > 0 {
		str = fmt.Sprintf("%ds ", allsecs)
	}
	return str
}
```
I think using a switch case would be better.

Also, there is a TimeAgo() in the component package... kindly make use of that. 
https://github.com/planetdecred/godcr/blob/269088483ecb35ae68550985dd6b39ae4df25591/ui/page/components/components.go#L280

song50119 replied:
TimeAgo is calculate base on timestamp, so it won't work like this

song50119 replied:
[this is how on dcrdata calculate](https://github.com/decred/dcrdata/blob/091b91b235e896a06ab086251892c1f6b6d7a366/explorer/templates.go#L200-L228)

Sirmorrison commented on wallet/commands.go:
```diff
@@ -284,8 +284,8 @@ func (wal *Wallet) GetTransaction(walletID int, txnHash string) {
-
-		txn, err := wall.GetTransactionRaw(hash[:])
+		txn, err := wall.GetTransactionRaw(hash.String())
+		log.Info(hash.String(), txn.Amount)
```
This change here is not part of the task are working on is it?

song50119 replied:
This PR have to update to latest version of dcrlibwallet, some function have to change to compatible with upstream

Sirmorrison commented on ui/page/tickets/overview.go:
```diff
@@ -125,9 +125,32 @@ func (pg *Page) ticketPriceSection(gtx layout.Context) layout.Dimensions {
+						return layout.Flex{Alignment: layout.Middle}.Layout(gtx,
+							layout.Rigid(func(gtx C) D {
+								tit := pg.Theme.Label(values.TextSize14, "Ticket Price")
+								tit.Color = pg.Theme.Color.Gray2
```
A better variable name will be `title` or `text` but not `tit`.

Sirmorrison commented on ui/page/tickets/overview.go:
```diff
@@ -125,9 +125,32 @@ func (pg *Page) ticketPriceSection(gtx layout.Context) layout.Dimensions {
+							}),
+						)
+					}
+					return pg.titleRow(gtx, leftWg, material.Switch(pg.Theme.Base, pg.autoPurchaseEnabled).Layout)
```
use decredmaterial Switch as it is customized to the project. You can see example usage on the settings page.

Commit:
Update duration till next ticket price
- Use decredmaterial Switch for autoPurchaseEnabled setting
- Remove TimeIcon and use exist TimerIcon

Sirmorrison commented on ui/page/tickets/overview.go:
```diff
@@ -52,7 +51,7 @@ func NewTicketPage(l *load.Load) *Page {
 		purchaseTicket:      l.Theme.Button(new(widget.Clickable), "Purchase"),
 
-		autoPurchaseEnabled: new(widget.Bool),
+		autoPurchaseEnabled: l.Theme.Switch(),
```
the `autoPurchaseEnabled` feature seem not to have been implemented. Kindly disable the button toggle.
`pg.autoPurchaseEnabled.Disabled()` in the `OnResume()` method

song50119 replied:
Would be great if create new PR for that function

Sirmorrison replied:
Switch widget disabled() already exist
https://github.com/planetdecred/godcr/blob/1dea6510eb61864cdfc28fe4a74973411a61a78e/ui/page/privacy_page.go#L56

Sirmorrison replied:
> the `autoPurchaseEnabled` feature seem not to have been implemented. Kindly disable the button toggle.
> `pg.autoPurchaseEnabled.Disabled()` in the `OnResume()` method

<img width="566" alt="image" src="https://user-images.githubusercontent.com/27733432/131457788-3af07dc2-9ff8-4c1e-b7c4-f450705cf731.png">

What was requested was to have the auto purchase toggle disabled not removed.
Kindly add it back as it is a vital part of the UI... all you need to so is the following.

```
func (pg *Page) OnResume() {
	pg.ticketPrice = dcrutil.Amount(pg.WL.TicketPrice()).String()
	go pg.WL.GetVSPList()
         // TODO: automatic ticket purchase functionality
        pg.autoPurchaseEnabled.Disabled()
}
```

Commit:
Disable auto purchase switch button
